internal/querycoordv2/meta/resource_group.go

```diff
@@ -176,28 +257,29 @@ func (rg *ResourceGroup) GetMeta() *querypb.ResourceGroup {
 // Snapshot return a snapshot of resource group.
 func (rg *ResourceGroup) Snapshot() *ResourceGroup {
 	return &ResourceGroup{
-		name:  rg.name,
-		nodes: rg.nodes.Clone(),
-		cfg:   rg.GetConfigCloned(),
+		name:    rg.name,
+		nodes:   rg.nodes.Clone(),
+		cfg:     rg.GetConfigCloned(),
+		nodeMgr: rg.nodeMgr,
 	}
 }
 
 // MeetRequirement return whether resource group meet requirement.
 // Return error with reason if not meet requirement.
 func (rg *ResourceGroup) MeetRequirement() error {
 	// if len(node) is less than requests, new node need to be assigned.
```

Comment:
Use RedundantNumOfNodes() > 0 || MissingNumOfNodes() > 0 to keep the consistency between api.